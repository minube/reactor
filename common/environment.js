import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
