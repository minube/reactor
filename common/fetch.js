const DEFAULT_HEADERS = {
  'Content-Type': 'application/json',
  'X-Requested-With': 'XMLHttpRequest',
};

export default async (endpoint, props = {}) => {
  const { headers, method = 'GET', ...others } = props;

  return new Promise((resolve, reject) => {
    fetch(endpoint, {
      headers: { ...DEFAULT_HEADERS, ...headers },
      method,
      ...others,
    })
      .then(async (response) => {
        const json = await response.json();

        if (response.status >= 400) reject({ code: response.status, message: json.message }); // eslint-disable-line
        else resolve(json);
      }).catch(({ message = 'ERROR_CONNECTION', response } = {}) => {
        reject({ // eslint-disable-line
          code: response ? response.status : undefined,
          message,
        });
      });
  });
};
