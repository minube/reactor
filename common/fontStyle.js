import screenSize from './screenSize';
import THEME from './theme';

const { FONT, UNIT } = THEME;

export default (
  width,
  {
    tiny, small, large, title,
  },
) => {
  const {
    TINY, PHONE, TABLET, SMALL, REGULAR, LARGE,
  } = screenSize(width);

  let fontSize = REGULAR || LARGE ? FONT.SIZE.REGULAR : FONT.SIZE.SMALL;

  if (tiny) fontSize = REGULAR || LARGE ? FONT.SIZE.TINY : UNIT * 1.1;
  else if (small) fontSize = REGULAR || LARGE ? FONT.SIZE.SMALL : FONT.SIZE.TINY;
  else if (large) {
    fontSize = FONT.SIZE.LARGE;
    if (TINY) fontSize = FONT.SIZE.SMALL;
    if (PHONE || SMALL) fontSize = FONT.SIZE.REGULAR;
    if (TABLET || REGULAR) fontSize = UNIT * 1.8;
  } else if (title) {
    fontSize = UNIT * 4.8;
    if (TINY || PHONE || TABLET) fontSize = UNIT * 2.4;
    if (SMALL || REGULAR) fontSize = UNIT * 3;
  }

  return {
    fontSize,
    lineHeight: fontSize * (TINY || SMALL ? 1.2 : 1.3),
  };
};
