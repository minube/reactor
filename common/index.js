import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
