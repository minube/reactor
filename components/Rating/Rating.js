import { array, bool, number, oneOfType, string } from 'prop-types';
import React from 'react';
import { StyleSheet, View } from 'react-native';

import Text from '../Text';
import styles from './Rating.style';

const Rating = ({
  caption, color, count, showValue, style, textColor, value, ...inherit
}) => (
  <View style={styles.container}>
    { [...Array(5).keys()].map(rate => (
      <Text {...inherit} primary color={color} key={rate} style={StyleSheet.flatten([styles.rate, style])}>
        {rate < Math.floor(value) ? '★' : '☆'}
      </Text>)) }
    { value > 0 && showValue &&
      <Text bold lighten tiny color={textColor} style={style}>{` ${value}/5`}</Text> }
    { count > 0 &&
      <Text lighten tiny color={textColor} style={style}>
        {' (' + `${count} ${caption}`.trim() + ')'}
      </Text> }
  </View>
);

Rating.propTypes = {
  caption: string,
  color: string,
  count: number,
  showValue: bool,
  style: oneOfType([array, number]),
  textColor: string,
  value: number,
};

Rating.defaultProps = {
  caption: '',
  color: undefined,
  count: 0,
  showValue: true,
  style: [],
  textColor: undefined,
  value: 0,
};

export default Rating;
