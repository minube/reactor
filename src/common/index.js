import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
