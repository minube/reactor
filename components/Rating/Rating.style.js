import { StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const { SPACE } = THEME;

export default StyleSheet.create({
  container: LAYOUT.STYLE.ROW,

  rate: {
    marginRight: SPACE.XS,
  },
});
