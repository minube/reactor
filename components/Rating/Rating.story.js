import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, color, number } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import Rating from './Rating';

storiesOf('✅ Rating', module)
  .add('default', () => (
    <Rating />
  ))
  .add('value', () => (
    <Rating value={3.5} />
  ))
  .add('showValue', () => (
    <Rating value={3.5} showValue />
  ))
  .add('count', () => (
    <Rating value={4} count={32} />
  ))
  .add('caption', () => (
    <Rating value={4} count={32} caption="valoraciones" />
  ))
  .add('color', () => (
    <Rating value={4} color={THEME.COLOR.SECONDARY} />
  ))
  .add('textColor', () => (
    <Rating count={32} value={4} color={THEME.COLOR.SECONDARY} textColor={THEME.COLOR.PRIMARY} />
  ))
  .add('style', () => (
    <Rating count={32} style={{ fontSize: 20, lineHeight: 20, color: 'orange' }} />
  ))
  .add('🏀 Playground', () => (
    <Rating
      color={color('color', THEME.COLOR.TEXT)}
      textColor={color('textColor', THEME.COLOR.TEXT_LIGHTEN)}
      count={number('count', 128)}
      showValue={boolean('showValue', true)}
      value={number('value', 3)}
    />
  ));

