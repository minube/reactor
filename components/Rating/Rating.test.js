import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Rating from './Rating';

describe('<Rating>', () => {
  it('renders', () => {
    const tree = renderer.create(<Rating />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<Rating value={3.5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {showValue}', () => {
    const tree = renderer.create(<Rating value={3.5} showValue />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {count}', () => {
    const tree = renderer.create(<Rating count={32} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {caption}', () => {
    const tree = renderer.create(<Rating caption="valoraciones" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Rating value={3.5} color={THEME.COLOR.ALTERNATIVE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {textColor}', () => {
    const tree = renderer.create(<Rating value={3.5} textColor={THEME.COLOR.ALTERNATIVE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Rating value={3.5} style={{ fontSize: 20, lineHeight: 20, color: 'orange' }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
