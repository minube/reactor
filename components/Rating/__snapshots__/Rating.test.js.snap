// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating> renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
</View>
`;

exports[`<Rating> when {caption} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
</View>
`;

exports[`<Rating> when {color} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "#424242",
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "#424242",
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "#424242",
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "#424242",
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "#424242",
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 14,
          "lineHeight": 18,
        },
        false,
        Object {},
      ]
    }
  >
     3.5/5
  </Text>
</View>
`;

exports[`<Rating> when {count} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 14,
          "lineHeight": 18,
        },
        false,
        Object {},
      ]
    }
  >
     (32)
  </Text>
</View>
`;

exports[`<Rating> when {showValue} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 14,
          "lineHeight": 18,
        },
        false,
        Object {},
      ]
    }
  >
     3.5/5
  </Text>
</View>
`;

exports[`<Rating> when {style} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "orange",
          "fontSize": 20,
          "lineHeight": 20,
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "orange",
          "fontSize": 20,
          "lineHeight": 20,
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "orange",
          "fontSize": 20,
          "lineHeight": 20,
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "orange",
          "fontSize": 20,
          "lineHeight": 20,
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "orange",
          "fontSize": 20,
          "lineHeight": 20,
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 14,
          "lineHeight": 18,
        },
        false,
        Object {
          "color": "orange",
          "fontSize": 20,
          "lineHeight": 20,
        },
      ]
    }
  >
     3.5/5
  </Text>
</View>
`;

exports[`<Rating> when {textColor} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 14,
          "lineHeight": 18,
        },
        false,
        Object {
          "color": "#424242",
        },
      ]
    }
  >
     3.5/5
  </Text>
</View>
`;

exports[`<Rating> when {value} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ★
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "marginRight": 2.5,
        },
      ]
    }
  >
    ☆
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 14,
          "lineHeight": 18,
        },
        false,
        Object {},
      ]
    }
  >
     3.5/5
  </Text>
</View>
`;
